easybuild/easyblocks/b/boost.py

```diff
@@ -224,57 +223,65 @@ def build_step(self):
-            bjamoptions += " --with-python"
+            self.bjamoptions += " --with-python"
+
+        self.log.info("Building boost with single and multi threading")
+        self.bjamoptions += " threading=single,multi --layout=tagged"
 
         if self.cfg['boost_mpi']:
             self.log.info("Building boost_mpi library")
-            self.build_boost_variant(bjamoptions + " --with-mpi", paracmd)
+            mpi_bjamoptions = " --with-mpi"
+            self.build_boost_variant(self.bjamoptions + mpi_bjamoptions, self.paracmd)
 
-        if self.cfg['boost_multi_thread']:
-            self.log.info("Building boost with multi threading")
-            self.build_boost_variant(bjamoptions + " threading=multi --layout=tagged", paracmd)
+        self.log.info("Building boost libraries")
+        # build with specified options
+        cmd = "%s ./%s %s %s %s" % (
+            self.cfg['prebuildopts'], self.bjamcmd, self.bjamoptions, self.paracmd, self.cfg['buildopts'])
+        run_cmd(cmd, log_all=True, simple=True)
 
-        # if both boost_mpi and boost_multi_thread are enabled, build boost mpi with multi-thread support
-        if self.cfg['boost_multi_thread'] and self.cfg['boost_mpi']:
-            self.log.info("Building boost_mpi with multi threading")
-            extra_bjamoptions = " --with-mpi threading=multi --layout=tagged"
-            self.build_boost_variant(bjamoptions + extra_bjamoptions, paracmd)
+    def install_step(self):
+        """Install Boost by copying files to install dir."""
 
-        # install remainder of boost libraries
+        # install boost libraries
         self.log.info("Installing boost libraries")
 
         cmd = "%s ./%s %s install %s %s" % (
-            self.cfg['preinstallopts'], self.bjamcmd, bjamoptions, paracmd, self.cfg['installopts'])
+            self.cfg['preinstallopts'], self.bjamcmd, self.bjamoptions, self.paracmd, self.cfg['installopts'])
         run_cmd(cmd, log_all=True, simple=True)
 
-    def install_step(self):
-        """Install Boost by copying files to install dir."""
-
         self.log.info("Copying %s to installation dir %s", self.objdir, self.installdir)
         if self.cfg['only_python_bindings'] and 'Python' in self.cfg['multi_deps'] and self.iter_idx > 0:
             self.log.info("Main installation should already exist, only copying over missing Python libraries.")
-            copy(glob.glob(os.path.join(self.objdir, 'lib', 'libboost_python*')), os.path.join(self.installdir, 'lib'))
+            copy(glob.glob(os.path.join(self.objdir, 'lib', 'libboost_python*')), os.path.join(self.installdir, 'lib'),
+                 symlinks=True)
         else:
-            copy(glob.glob(os.path.join(self.objdir, '*')), self.installdir)
+            copy(glob.glob(os.path.join(self.objdir, '*')), self.installdir, symlinks=True)
+
+        # Link tagged multi threaded libs as the default libs
+        lib_glob = 'lib*-mt-*.*'
+        mt_replace = re.compile(r'-mt[^.]*\.')
```

Comment:
```suggestion
        lib_glob = 'lib*-mt*.*'
        mt_replace = re.compile(r'-[^.]*\.')
```

Older boost had `libboost_system-mt.so` or similar. So no second dash. Also the regex can be changed to really remove all tags, i.e. everything starting at the dash and until the dot. No need to rely on mt

Reply:
Have never seen a build that has -mt.so, but yeah